config/confighttp/confighttp_test.go

```diff
@@ -89,9 +105,14 @@ func TestAllHTTPClientSettings(t *testing.T) {
 				return
 			}
 			assert.NoError(t, err)
-			transport := client.Transport.(*http.Transport)
```

Comment:
This type assertion didn't work when I added Compression: "gzip" since this transport is no more *http.Transport but middleware.CompressRoundTripper. So I chaned the test like the right side.